### DISCUSSION TITLE
Help: How to know the log level stablished by compiler?

### DISCUSSION THREAD

**Discussion by ricard2k** [espressif/arduino-esp32#8259] [General]
esp_log_level_get("*") always returns ESP_LOG_ERROR. unless you define other loglevel in tools/core log level. 

`#include <Arduino.h>

void setup() {
  Serial.begin(115200, SERIAL_8N1);
  Serial.print("ESP32 test get loglevel.\n");
  switch (esp_log_level_get("*")) {
    case ESP_LOG_NONE:
      Serial.print("ESP_LOG_NONE\n");
      break;
    case ESP_LOG_ERROR:
      Serial.print("ESP_LOG_ERROR\n");
      break;
    case ESP_LOG_WARN:
      Serial.print("ESP_LOG_WARN\n");
      break;
    case ESP_LOG_INFO:
      Serial.print("ESP_LOG_INFO\n");
      break;
    case ESP_LOG_DEBUG:
      Serial.print("ESP_LOG_DEBUG\n");
      break;
    case ESP_LOG_VERBOSE:
      Serial.print("ESP_LOG_VERBOSE\n");
  }

}

void loop() {
  // put your main code here, to run repeatedly:

}`

How to get the real log level?

**Comment by mrengineer7777** [comment]
Arduino doesn't use IDF logging. You can determine the log level with CORE_DEBUG_LEVEL. See https://github.com/espressif/arduino-esp32/blob/master/cores/esp32/esp32-hal-log.h.

**Comment by ricard2k** [comment]
For other people reference, The log Level is defined in ARDUHAL_LOG_LEVEL macro and the leves has been redefined as:

- ARDUHAL_LOG_LEVEL_NONE
- ARDUHAL_LOG_LEVEL_ERROR
- ARDUHAL_LOG_LEVEL_WARN
- ARDUHAL_LOG_LEVEL_INFO
- ARDUHAL_LOG_LEVEL_DEBUG
- ARDUHAL_LOG_LEVEL_VERBOSE


Not the best example about how to use the precompiler macros, but easy to test:
`
#include <Arduino.h>

void setup() {
  Serial.begin(115200, SERIAL_8N1);
  Serial.print("ESP32 test get loglevel.\n");
  switch (ARDUHAL_LOG_LEVEL) {
    case ARDUHAL_LOG_LEVEL_NONE:
      Serial.print("ARDUHAL_LOG_LEVEL_NONE\n");
      break;
    case ARDUHAL_LOG_LEVEL_ERROR:
      Serial.print("ARDUHAL_LOG_LEVEL_ERROR\n");
      break;
    case ARDUHAL_LOG_LEVEL_WARN:
      Serial.print("ARDUHAL_LOG_LEVEL_WARN\n");
      break;
    case ARDUHAL_LOG_LEVEL_INFO:
      Serial.print("ARDUHAL_LOG_LEVEL_INFO\n");
      break;
    case ARDUHAL_LOG_LEVEL_DEBUG:
      Serial.print("ARDUHAL_LOG_LEVEL_DEBUG\n");
      break;
    case ARDUHAL_LOG_LEVEL_VERBOSE:
      Serial.print("ARDUHAL_LOG_LEVEL_VERBOSE\n");
  }

}

void loop() {
  // put your main code here, to run repeatedly:

}
`
